frontend/src/pages/modelCatalog/screens/ModelDetailsPage.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { useParams } from 'react-router';
+import { Button, Divider } from '@patternfly/react-core';
+import ApplicationsPage from '~/pages/ApplicationsPage';
+import TitleWithIcon from '~/concepts/design/TitleWithIcon';
+import { ProjectObjectType, typedEmptyImage } from '~/concepts/design/utils';
+import { conditionalArea, SupportedArea } from '~/concepts/areas';
+import { ModelCatalogContext } from '~/concepts/modelCatalog/context/ModelCatalogContext';
+import { CatalogModel } from '~/concepts/modelCatalog/types';
+import EmptyModelCatalogState from '~/pages/modelCatalog/EmptyModelCatalogState';
+import { decodeParams, findModelFromModelCatalogSources } from '~/pages/modelCatalog/utils';
+import ModelDetailsView from './ModelDetailsView';
+
+const ModelDetailsPage: React.FC = conditionalArea(
+  SupportedArea.MODEL_CATALOG,
+  true,
+)(() => {
+  const params = useParams();
```

Comment:
We can use generics here to narrow the type of these params rather than relying on them being `Record<string, string>`.

If you import your new type `ModelDetailsRouteParams` here and pass it to useParams:
```ts
const params = useParams<ModelDetailsRouteParams>();
```
then `params` will have the type `Partial<ModelDetailsRouteParams>`. You can then use that type for all the functions you're passing these params to instead of `Record<string, string>`.

Note that if all the properties of `ModelDetailsRouteParams` are optional, you should just be able to have your functions take that instead of `Partial<ModelDetailsRouteParams>`, because the two will be equivalent.